pkg/controller/lambda/function/setup.go

```diff
@@ -0,0 +1,499 @@
+		return false, nil
+	}
+
+	addTags, removeTags := diffTags(cr.Spec.ForProvider.Tags, obj.Tags)
+	return len(addTags) == 0 && len(removeTags) == 0, nil
+
+}
+
+// isUpToDateEnvironment checks if FunctionConfiguration EnvironmentResponse Variables are up to date
+func isUpToDateEnvironment(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
+	envVars := map[string]*string{}
+	awsVars := map[string]*string{}
+	if !reflect.ValueOf(cr.Spec.ForProvider.Environment).IsZero() &&
+		!reflect.ValueOf(cr.Spec.ForProvider.Environment.Variables).IsZero() {
+		envVars = cr.Spec.ForProvider.Environment.Variables
+	}
+	if !reflect.ValueOf(obj.Configuration.Environment).IsZero() &&
+		!reflect.ValueOf(obj.Configuration.Environment.Variables).IsZero() {
+		awsVars = obj.Configuration.Environment.Variables
+	}
+	// Check if counts match which should handle nil/empty
+	if len(envVars) != len(awsVars) {
+		return false
+	}
+	// Check if actual values match, and only DeepEqual if there are actual values
+	if len(envVars) > 0 &&
+		!reflect.DeepEqual(envVars, awsVars) {
+		return false
+	}
+	return true
+}
+
+func isUpToDateFileSystemConfigs(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
+	var fileSystemConfigs []*svcsdk.FileSystemConfig
+	var awsFileSystemConfigs []*svcsdk.FileSystemConfig
+	if !reflect.ValueOf(cr.Spec.ForProvider.FileSystemConfigs).IsZero() {
+		for _, v := range cr.Spec.ForProvider.FileSystemConfigs {
+			fileSystemConfigs = append(fileSystemConfigs, &svcsdk.FileSystemConfig{
+				Arn:            v.ARN,
+				LocalMountPath: v.LocalMountPath,
+			})
+		}
+	}
+	if !reflect.ValueOf(obj.Configuration.FileSystemConfigs).IsZero() {
+		awsFileSystemConfigs = obj.Configuration.FileSystemConfigs
+	}
+	// Check if counts match which should handle nil/empty
+	if len(fileSystemConfigs) != len(awsFileSystemConfigs) {
+		return false
+	}
+	// Check if actual values match, and only DeepEqual if there are actual values
+	if len(fileSystemConfigs) > 0 &&
+		!reflect.DeepEqual(fileSystemConfigs, awsFileSystemConfigs) {
+		return false
+	}
+	return true
+}
+
+func isUpToDateCode(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// GetFunctionOutput returns
+	// Code *FunctionCodeLocation `type:"structure"`
+	// which does not map to
+	// Code *FunctionCode `type:"structure" required:"true"`
+	// which is used when creating the function.
+	// We can't currently properly implement a comparison
+	return true
+}
+
+func isUpToDateLayers(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// The generator is creating layers with type of
+	// 	Layers []*string `json:"layers,omitempty"`
+	// Instead of []*svcsdk.Layer
+	// We can't properly implement a comparison until that is fixed
+	return true
+}
+
+func isUpToDateTracingConfig(cr *svcapitypes.Function, obj *svcsdk.GetFunctionOutput) bool {
+	// Handle nil pointer refs
```

Comment:
You can make sure both pointers are non-nil before doing the string comparison. If any one or both of them are nil, then it can return `false`.